Rakefile

```diff
@@ -23,7 +33,116 @@ task :travis do
   Rake::Task['spec'].invoke
 end
 
+module HarnessOptions
+  defaults = {
+      :tests  => ['tests'],
+      :log_level => 'debug',
+      :preserve_hosts => 'onfail',
+  }
+
+  DEFAULTS = defaults
+
+  def self.get_options(file_path)
+    puts "Attempting to merge config file: #{file_path}"
+    if File.exists? file_path
+      options = eval(File.read(file_path), binding)
+    else
+      puts "No options file found at #{File.expand_path(file_path)}... skipping"
+    end
+    options || {}
+  end
+
+  def self.get_local_options
+    get_options("./local_options.rb")
+  end
+
+  def self.final_options(intermediary_options = {})
+    local_overrides = get_local_options
+    final_options = DEFAULTS.merge(intermediary_options)
+    final_options.merge!(local_overrides)
+    return final_options
+  end
+
+end
+
+def hosts_file_env
+  ENV['BEAKER_HOSTS']
+end
+
+def hosts_opt(use_preserved_hosts=false)
+  if use_preserved_hosts
+    "--hosts=#{HOSTS_PRESERVED}"
+  else
+    if hosts_file_env
+      "--hosts=#{hosts_file_env}"
+    else
+      "--hosts=tmp/#{HOSTS_FILE}"
+    end
+  end
+end
+
+def agent_target
+  ENV['TEST_TARGET'] || 'redhat7-64af'
+end
+
+def master_target
+  ENV['MASTER_TEST_TARGET'] || 'redhat7-64default.mdcal'
+end
+
+def test_targets
+  ENV['LAYOUT'] || "#{master_target}-#{agent_target}"
+end
+
+HOSTS_FILE = "#{test_targets}-#{SecureRandom.uuid}.yaml"
+
+def pe_dir
+  ENV['PE_DIR'] || "http://neptune.puppetlabs.lan/archives/releases/3.7.2"
```

Comment:
This was based on what's currently used in the Jenkins runs for this pre-suite/test. It is difficult to know what to include as defaults (we are also using beaker-hostgenerator if no hosts file is defined and by default it sets hypervisor to vmpooler) as the ticket was originally raised by a Puppet dev who was having to go and hunt through Jenkins jobs and amend them/run them to check his local changes to Beaker. To make it easy for internal people to just run 'rake acceptance' vs community members having to set extra variables/ include extra files is the dilemma.  We could just remove the puppetpe task altogether and have puppetgit (puppet git install) as part of the acceptance run.